File: src/FindThatBook.Api/Models/Book.cs
namespace FindThatBook.Api.Models;

public sealed record Book(
    string Title,
    string Author,
    int? FirstPublishYear,
    string Description,
    string? BookKey,
    string? BookUrl,
    int? CoverId,
    string? CoverImageUrl,
    string Explanation,
    int? Score = null)
{
    public IReadOnlyList<BookAuthor> Authors { get; init; } = [];
}

public sealed record BookAuthor(
    string? AuthorKey,
    string Name,
    string? Role,
    bool IsPrimary,
    string Evidence);

File: src/FindThatBook.Web/src/models/Book.ts
export type Book = {
  title: string
  author: string
  firstPublishYear: number | null
  description: string
  bookKey: string | null
  bookUrl: string | null
  coverId: number | null
  coverImageUrl: string | null
  explanation: string
  score: number
  authors: BookAuthor[]
}

export type BookAuthor = {
  authorKey: string | null
  name: string
  role: string | null
  isPrimary: boolean
  evidence: 'canonicalWork' | 'searchResult'
}
